src/std/net/json-rpc.ss

```diff
@@ -35,30 +35,35 @@
   (only-in :std/text/json trivial-json-object->class JSON json-symbolic-keys
            bytes->json-object json-object->bytes json-object->string))
 
-(defstruct (json-rpc-error <error>) ()
+(defclass (JSON-RPCError IOError) ()
 ;;  (code    ;; SInt16
 ;;   message ;; String
 ;;   data)   ;; (Maybe Bytes)
-  transparent: #t constructor: :init!)
-(defmethod {:init! json-rpc-error}
+  transparent: #t)
+(defmethod {:init! JSON-RPCError}
   (lambda (self what: (what "JSON RPC error") where: (where 'json-rpc)
            code: code ;; SInt16
            message: message ;; String
            data: (data (void))) ;; (Maybe Bytes)
     (def irritants [code message data])
-    (struct-instance-init! self what irritants where)))
```

Comment:
The "what" should be the message: and the irritants in general is a leftover of unstructured exceptions and should be replaced by the fields of the exception, or something computed from them.